app/utils/__tests__/gameLogic.test.ts

```diff
@@ -1,7 +1,32 @@
 import { getInitialGameState } from '../gameLogic';
 import { GameState } from '../../types/game';
+import * as deckUtils from '../deckUtils';
+import * as gameUtils from '../gameUtils';
+
+jest.mock('../deckUtils', () => ({
+  createDeck: jest.fn(),
+  dealCards: jest.fn()
+}));
+
+jest.mock('../gameUtils', () => ({
+  determineWinner: jest.fn(),
+  hasFlor: jest.fn()
```

Comment:
I don't understand why we are mocking those functions. Specially `hasFlor` that is the one implementing the feature under test.

I'm ok with mocking `dealCards` since it has random behavior, but mocking `createDeck` seems overkill to me.